ts/features/idpay/wallet/utils/hooks.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+
+import { useFocusEffect } from "@react-navigation/native";
+import { useSelector } from "react-redux";
+import { useIODispatch } from "../../../../store/hooks";
+import { idPayInitiativesFromInstrumentGet } from "../store/actions";
+import {
+  idPayAreInitiativesFromInstrumentErrorSelector,
+  idPayEnabledInitiativesFromInstrumentSelector
+} from "../store/reducers";
+
+export const useIDPayInitiativesFromInstrument = (idWallet: string) => {
+  const dispatch = useIODispatch();
+  const namedInitiativesList = useSelector(
+    idPayEnabledInitiativesFromInstrumentSelector
+  );
+  const areInitiativesInError = useSelector(
+    idPayAreInitiativesFromInstrumentErrorSelector
+  );
+  React.useEffect(() => {
+    // required to clear any old data
+    dispatch(
+      idPayInitiativesFromInstrumentGet.request({
+        idWallet,
+        isRefreshCall: false
+      })
+    );
+  }, [idWallet, dispatch]);
+  useFocusEffect(() => {
+    // will be removed in #IODPAY-207 in favor of saga logic
+    const timer = setInterval(
+      () => {
+        dispatch(
+          idPayInitiativesFromInstrumentGet.request({
+            idWallet,
+            isRefreshCall: true
+          })
+        );
+      },
+      areInitiativesInError ? 6000 : 3000
+    );
+    return () => clearInterval(timer);
+  });
```

Comment:
```suggestion
  useFocusEffect(() => {

dispatch(
      idPayInitiativesFromInstrumentGet.request({
        idWallet,
        isRefreshCall: false
      })
    );

    // will be removed in #IODPAY-207 in favor of saga logic
    const timer = setInterval(
      () => {
        dispatch(
          idPayInitiativesFromInstrumentGet.request({
            idWallet,
            isRefreshCall: true
          })
        );
      },
      areInitiativesInError ? 6000 : 3000
    );
    return () => clearInterval(timer);
  });
```

Hooks could be merged. What do you think?

Reply:
I have tried merging them, and it resulted in a constant re-fetch. have you tested it in your editor? I may be missing something

Reply:
Just tested. Works pretty well!